tests/functional/ipset_controller_test.go

```diff
@@ -369,11 +369,22 @@ var _ = Describe("IPSet controller", func() {
 
 			Eventually(func(g Gomega) {
 				g.Expect(k8sClient.Get(ctx, ipSetName, instance)).Should(Succeed())
+				ctlplaneNetwork := instance.Status.Reservation[0].Network
+				ctlplaneFound := false
 				for i := 0; i < len(ipSetNetworks); i++ {
 					// first assert that the instance networks are in the same order as we specified
 					g.Expect(instance.Spec.Networks[i].Name).To(Equal(ipSetNetworks[i].Name))
 					// then assert that the reservation networks are in the same order
-					g.Expect(instance.Spec.Networks[i].Name).To(Equal(instance.Status.Reservation[i].Network))
+					// other than ctlplane network being the first one.
+					if ipSetNetworks[i].Name == ctlplaneNetwork {
+						ctlplaneFound = true
+						continue
+					}
+					if ctlplaneFound {
+						g.Expect(instance.Spec.Networks[i].Name).To(Equal(instance.Status.Reservation[i].Network))
```

Comment:
maybe we verify that `i`  must be `0` for the ctlplane case? `g.Expect(i).To(Equal(0))`

Reply:
The test does that to identify the ctlplane network in L372, right?

Reply:
right, with L372 and the condition in L379 it would be the same what I commented, but I don't think we have a test that the network[0] ServiceNetwork is "ctlplane", or do we? maybe we should have a validation that `instance.Status.Reservation[0].ServiceNetwork == "ctlplane"`?